src/components/ui/user-dropdown.tsx

```diff
@@ -0,0 +1,126 @@
+import ClickAwayListener from "@mui/material/ClickAwayListener";
+import Fade from "@mui/material/Fade";
+import { styled } from "@mui/material/styles";
+import { useRef, useState } from "react";
+import type { MouseEvent } from "react";
+
+import { Avatar } from "@/components/ui/avatar";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import { Divider } from "@/components/ui/divider";
+import { MenuList } from "@/components/ui/menu-list";
+import { Paper } from "@/components/ui/paper";
+import { Popper } from "@/components/ui/popper";
+import { Typography } from "@/components/ui/typography";
+
+const BadgeContentSpan = styled("span")({
+  width: 8,
+  height: 8,
+  borderRadius: "50%",
+  cursor: "pointer",
+  backgroundColor: "var(--mui-palette-success-main)",
+  boxShadow: "0 0 0 2px var(--mui-palette-background-paper)",
+});
+
+type UserDropdownProps = {
+  user: {
+    name: string;
+    email: string;
+    image: string;
+  };
+};
+
+export const UserDropdown = ({ user }: UserDropdownProps) => {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  const handleDropdownOpen = () => {
+    !open ? setOpen(true) : setOpen(false);
```

Comment:
これってVuexyのコピペ？
だとしたら結構コード微妙っすね。。

```suggestion
    setOpen((prev) => !prev);
```

Reply:
関数名もhandleDropdownToggleみたいな命名が正しい

Reply:
すんません。流石にこれは直しておくべきでした 🙏 
これはOpenの関数で使いたいので

```
const handleDropdownOpen = () => {
    setOpen(true);
  };
```

こうしておきました！